pkg/cloud/azure/actuators/machine_scope_test.go

```diff
@@ -324,3 +387,71 @@ func TestNewMachineScope(t *testing.T) {
 		}
 	}
 }
+
+func TestGetCloudEnvironment(t *testing.T) {
+	testCases := []struct {
+		name                string
+		client              client.Client
+		expectedError       bool
+		expectedEnvironment string
+	}{
+		{
+			name:          "fail when infrastructure is missing",
+			client:        controllerfake.NewFakeClient(),
+			expectedError: true,
+		},
+		{
+			name: "when cloud environment is missing default to Azure Public Cloud",
+			client: controllerfake.NewFakeClient(&configv1.Infrastructure{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: globalInfrastuctureName,
+				},
+				Status: configv1.InfrastructureStatus{
+					PlatformStatus: &configv1.PlatformStatus{
+						Azure: &configv1.AzurePlatformStatus{}},
+				},
+			}),
+			expectedEnvironment: string(configv1.AzurePublicCloud),
+		},
+		{
+			name: "when cloud environment is present return it",
+			client: controllerfake.NewFakeClient(&configv1.Infrastructure{
```

Comment:
can we include a case for when PlatformStatus is nil. I think that will make the func panic as it is now.